File: src/main/java/com/indiepost/model/Image.java
package com.indiepost.model;

import com.indiepost.enums.Types.ImageSize;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.io.Serializable;

/**
 * Created by jake on 8/15/16.
 */
@Entity
@Table(name = "Images")
public class Image implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    @Size(min = 2, max = 120)
    private String filePath;

    @Column(nullable = false)
    @Size(min = 2, max = 120)
    private String fileName;

    @Column(nullable = false)
    private int width;

    @Column(nullable = false)
    private int height;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private ImageSize sizeType;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public ImageSize getSizeType() {
        return sizeType;
    }

    public void setSizeType(ImageSize sizeType) {
        this.sizeType = sizeType;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }
}

File: src/main/kotlin/com/indiepost/model/Image.kt
package com.indiepost.model

import com.indiepost.enums.Types.ImageSize
import java.io.Serializable
import javax.persistence.*
import javax.validation.constraints.Size

/**
 * Created by jake on 8/15/16.
 */
@Entity
@Table(name = "Images")
data class Image(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null,

        @Column(nullable = false)
        @Size(min = 2, max = 120)
        var filePath: String? = null,

        @Column(nullable = false)
        @Size(min = 2, max = 120)
        var fileName: String? = null,

        @Column(nullable = false)
        var width: Int = 0,

        @Column(nullable = false)
        var height: Int = 0,

        @Column(nullable = false)
        @Enumerated(EnumType.STRING)
        var sizeType: ImageSize? = null,

        @Column(name = "imageSetId", updatable = false, insertable = false)
        var imageSetId: Long? = null
) : Serializable {
    companion object {
        private const val serialVersionUID = -6417761322329861824L
    }
}
